docs/tutorial/flow-nft-marketplace.md

```diff
@@ -0,0 +1,1804 @@
+> it in a function. Then I repeat.
+>
+> This makes perfect sense. The more code we write, the harder it is
+> to debug and go back to fix something. It is always better to write > less code in general.
+>
+> Here, I would include `console.log()` in each event handler, then
+> test clicking the form to make sure the logged values are what I
+> expected before moving on.
+
+## Minting logic
+
+Here comes the interesting part. We will hook up the `Mint` button to actually mint a token based on user's input!
+
+Switching gears here, let's head over to `/src/flow/transaction` and create a new JavaScript file named `MintToken.tx.js`. This module will send the Cadence transaction `MintToken.cdc` similarly to how we have used Flow CLI to do so previously.
+
+Here I create a JavaScript module that interacts with each Cadence transaction or script and name it to reflect the Cadence code, appended by `.tx.js` or a transaction or `.sc.js` for a script. This is not a requirement and you're free to name them however you want.
+
+Since there will be quite a lot going on, we will go slowly on this one. We will create a `mintToken` function that takes a `pet` object and does the following:
+
+1. Upload the metadata and image asset to NFT.storage, and retrieve the returned metadata that includes the [CID][ipfs-cid] of the data.
+2. Send a minting transaction with the metadata to Flow (in this case, the name, age, breed, and the CID of the data stored on IPFS).
+3. If successful, return the Flow transaction ID.
+
+
+Here we sketch up some placeholder functions to reflect the steps:
+
+```js
+
+// MintToken.tx.js
+
+async function mintToken(pet) {
+  let metadata = await uploadToStorage(pet);
+  let txId = await mintPet(metadata);
+  return txId;
+}
+
+/* We will fill in these functions next */
+
+async function uploadToStorage(pet) {
+    return {};
+}
+
+async function mintPet(metadata) {
+    return '';
+}
+
+```
+
+> **💡 Where to catch an error?**
+> Asynchronous functions which return a `Promise` are *fallable*. Conventionally, you should prepare for the worst
+> by catching the error. But the question is where to do this?
+> The general rule here is if a function does not know what to do with the error, don't catch it. Delegate this responsibility
+> to the caller who, hopefully, will know better. In the previous `mintToken` function, we didn't
+> handle the error because we wanted the calling function to handle it.
+> This way, you are not making things worse by complicating the code with unnecessary `try-catch` blocks that only obscure
+> the errors. Let the one who knows better deals with it.
+
+Next, we will fill in the body of `uploadToStorage` function. This is where you will need your API key from NFT.Storage:
+
+```js
+
+// Import these modules from nft.storage on top of the file.
+import { NFTStorage, File } from 'nft.storage';
+
+const API_KEY = "DROP_YOUR_API_KEY_HERE";
+
+// Initialize the NFTStorage client.
```

Comment:
```suggestion
// Initialize the NFTStorage client
```

Most of your other code doesn't have periods in comments. Let's choose one style for consistency's sake